runtime/src/main/java/org/corfudb/security/tls/ReloadableTrustManager.java

```diff
@@ -36,21 +36,21 @@ public ReloadableTrustManager(TrustStoreConfig trustStoreConfig) throws SSLExcep
 
     @Override
     public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
-        checkValidity(chain);
-
         reloadTrustStoreWrapper();
+        checkValidity(chain);
```

Comment:
Is there a specific reason why the order is swapped?

Reply:
The original order was incorrect as we need to reload the trust store before checking the validity of the certificates.